firedrake/extrusion_extend.py

```diff
@@ -0,0 +1,44 @@
+from firedrake import FunctionSpace, Function, DirichletBC
+
+__all__ = ['ExtrudedExtendFunction']
+
+def ExtrudedExtendFunction(mesh, base_mesh, fcn, extend_type=None):
```

Comment:
The `base_mesh` argument isn't needed -- you can access the base through `mesh._base_mesh`.